### ISSUE TITLE
Make ECDSA signature verification host call customizable

### ISSUE THREAD

**Issue by xlc** [smol-dot/smoldot#2016]
https://github.com/paritytech/smoldot/pull/2955 covers many host calls but not `ext_crypto_secp256k1_ecdsa_recover_version_2`, which is used by Moonbeam for their Ethereum signature verification. This means Chopsticks won't be able to mock their signature.

cc @albertov19 https://github.com/AcalaNetwork/chopsticks/issues/845

**Comment by tomaka** [comment]
This is a bit complicated, because smoldot simply calls [`libsecp256k1::recover`](https://docs.rs/libsecp256k1/latest/libsecp256k1/fn.recover.html), which verifies the signature and returns the public key both in the same function.

Since the `ext_crypto_secp256k1_ecdsa_recover_version_2` function returns that public key, if the signature is invalid it simply can't proceed.

`libsecp256k1` doesn't seem to have a way to pass a dummy signature and still recover the public key, and I don't know enough about secp256k1 to know whether this makes sense at all.

**Comment by tomaka** [comment]
Does it make sense for you if, when you're trying to mock an ECDSA signature, smoldot asks you to provide the associated public key?

**Comment by ermalkaleci** [comment]
@xlc  I don't think there's  a need to change anything on smoldot. Chopsticks mock signature is creating signature with 64 bytes while ecdsa is 65. I think that might be the issue. Checking now

**Comment by ermalkaleci** [comment]
my bad, seems like `RuntimeCall::SignatureVerification` is called at all

**Comment by xlc** [comment]
> Does it make sense for you if, when you're trying to mock an ECDSA signature, smoldot asks you to provide the associated public key?

yeah that’s fine. we can modify the mock signature format to include a public key
